### PR TITLE
chore: remove x86_64-darwin from CI

### PR DESCRIPTION
**Pull request by d4hines**
No one on the team is using x86_64-darwin, and
it's slowing down our CI. Let's remove it.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        system: [ubuntu, macos]
+        system: [ubuntu]
```
Comment (ulrikstrid) [inline .github/workflows/test.yml:15]: Can we change this to `ubuntu-latest` while we're at it?

**Review by ulrikstrid — APPROVED** [review]
Do we have to make it explicit that we don't officially support `x86_64-darwin` in the readme?

File: .github/workflows/test.yml (hunk 1 above)
Comment (d4hines) [reply .github/workflows/test.yml:15]: Isn't it already running on latest via `${{ matrix.system }}-latest` on the next line?

Comment (ulrikstrid) [reply .github/workflows/test.yml:15]: You are correct, missed that :facepalm:

File: flake.nix [hunk 2]
```diff
@@ -49,8 +49,7 @@
   outputs = { self, nixpkgs, flake-utils, nix-filter, dream2nix, ocaml-overlays
     , prometheus-web, tezos, json-logs-reporter }:
     let
-      supportedSystems =
-        [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" ];
```
Comment (ulrikstrid) [inline flake.nix:52]: I will try to setup my hydra for `aarch64-linux` soon so that we can actually test that workflow somewhere

**Comment by EduardoRFS** [comment]
This is not a good change, while no one is using x86_64-darwin a bunch of people are running arm64-darwin, which was tested in part by x86_64-darwin.

**Comment by EduardoRFS** [comment]
Also the only slow down of it was using an additional github actions instance, as it did not prevent merging.

**Comment by ulrikstrid** [comment]
@EduardoRFS there was a bunch of errors, both with esy (problem compiling OCaml?) and nix (installing the nix action have 403 frequently).
aarch64-darwin is tested by hydra, even though it's not official (yet) it's an actual test of the architecture.

**Comment by EduardoRFS** [comment]
@ulrikstrid If the reason to remove it was because it was failing randomly, that's an even worse excuse to be removing support to it, it was an evidence of something being broken.

So instead of fixing the problem on nix, that someone may hit we decided to just drop the architecture.

**Pull request merged**